### PR TITLE
[docs] Generate ToC for Cookbooks

### PR DESCRIPTION
**Pull request by jackkoenig**
Also put cookbooks in a better order for new users.

Figuring out this CSS was annoying, many thanks to @richardxia for help.

Before:

Screenshot 2024-01-29 at 9 06 55 PM

After:

Screenshot 2024-01-29 at 9 07 08 PM

I do kind of like how we didn't have the "group" bullets as links, but this is sooooo much easier to maintain.

We can also then decide to have more interesting ToC styling, but I at least wanted to make sure we could replicate the styling exactly (or at least, very very close).

### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

- Documentation or website-related

#### Desired Merge Strategy

- Squash

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.6.x`, `5.x`, or `6.x` depending on impact, API modification or big change: `7.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: docs/src/cookbooks/cookbook.md [hunk 1]
```diff
@@ -507,7 +478,7 @@ class CustomBundleFixed(elts: (String, Data)*) extends Record {
 }
 ```
 
-### How do I create a finite state machine (FSM)?
+## How do I create a finite state machine (FSM)?
```
Comment (jackkoenig) [inline docs/src/cookbooks/cookbook.md:481]: There were accidentally getting grouped under the `Bundles` heading. See screenshot below of the current version of this page (ie. the one live on the website right now). See how we manually didn't group them under Bundles, but the autogenerated sidebar does have them under it.

Screenshot 2024-01-29 at 9 25 45 PM

File: website/src/css/custom.css [hunk 2]
```diff
@@ -53,6 +53,20 @@ a:hover {
...
+ */
+div[class^='tableOfContentsInline'] ul {
+  padding-left: var(--ifm-list-left-padding);
+  list-style-type: revert;
+}
+
+div[class^='tableOfContentsInline'] li {
+  margin: revert;
+}
+div[class^='tableOfContentsInline'] li + li {
```
Comment (richardxia) [inline website/src/css/custom.css:67]: Might just be a question of personal taste, but I find it more readable to write this selector as

```suggestion
div[class^='tableOfContentsInline'] li:not(:first-child) {
```

They're technically not equivalent, but I'm assuming you are using the sibling selector (`+`) in order to select everything but the first child.

**Review by richardxia — APPROVED** [review]
LGTM

File: website/src/css/custom.css (hunk 2 above)
Comment (jackkoenig) [reply website/src/css/custom.css:67]: I will sanity check that it gives the same result, but I agree that is better.

Comment (jackkoenig) [reply website/src/css/custom.css]: Looks the same, thanks!

**Review by azidar — APPROVED** [review]

**Review by mwachs5 — APPROVED** [review]

**Pull request merged**